### DISCUSSION TITLE
OpenRoad generated spef file syntax error

### DISCUSSION THREAD

**Discussion by faisaladilquadery** [The-OpenROAD-Project/OpenROAD-flow-scripts#1573] [Q&A]
I was working on integrating an opensource design "tinyriscv" with an Instance count of 40000 with tsmc65 pdk in Openroad flow. The spef file that is generated during parasitics extraction of the flow seems to be getting syntax errors. What could be the solution to this ?

*The BCrules file for tsmc65 pdk was extracted by following the RC Extraction Flow

![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/97328d00-2a25-4358-9a16-b8ebefb482c8)

**Comment by maliberty** [comment]
Please post the contents of that line

**Reply by faisaladilquadery** [reply]
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/4cd2e1c9-f659-4932-9400-ef924708ed81)

I can notice the syntax error here, and if i fix this, error reads in another line ,and it keeps repeating
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/cc54e787-3a5e-44e0-8d3f-a7f8fbe71f6f)

**Reply by maliberty** [reply]
A little more context before and after would help as its hard to tell what part of the output this is.

**Reply by faisaladilquadery** [reply]
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/63a3ab4c-cecf-434b-add1-b898eac111c3)

**Reply by faisaladilquadery** [reply]
syntax errors are part of the name_map section 
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/fdf9b518-acd2-4140-81c1-4b5330f34c01)

**Reply by faisaladilquadery** [reply]
@maliberty 
I have managed to find the root cause of the issue, If there are multiple third brackets in the naming syntax fromt he netlist as like mentioned here :
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/73973f85-0580-4c67-8076-105e79feff6c)
Syntax errors are created sometimes (not all times) by OpenRCX. 
Therefore i have changed the naming syntax while synthesizing the gate level netlist. The spef generated after the change is as follows: 
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/140678965/4004d780-616c-468f-b770-5d497d566016)
I have changed the naming syntax of third bracketing multiple times with underscore. After this no more syntax errors were reported.
